src/query/anonymization.c

```diff
@@ -150,6 +151,197 @@ static void rewrite_to_anon_aggregator(Aggref *aggref, List *aid_refs, Oid fnoid
+  count_aggref->aggdirectargs = list_copy(source_aggref->aggdirectargs);
+  count_aggref->args = list_copy(source_aggref->args);
+  count_aggref->aggorder = list_copy(source_aggref->aggorder);
+  count_aggref->aggdistinct = source_aggref->aggdistinct;
+  count_aggref->aggfilter = source_aggref->aggfilter;
+  count_aggref->aggstar = false;
+  count_aggref->aggvariadic = source_aggref->aggvariadic;
+  count_aggref->aggkind = source_aggref->aggkind;
+  count_aggref->agglevelsup = source_aggref->agglevelsup;
+  count_aggref->aggsplit = source_aggref->aggsplit;
+  count_aggref->location = source_aggref->location;
+  return count_aggref;
+}
+
+static FuncExpr *make_cast_common(List *args, Oid location)
+{
+  FuncExpr *cast = makeNode(FuncExpr);
+  cast->funcretset = false;
+  cast->funcvariadic = false;
+  cast->funcformat = COERCE_EXPLICIT_CAST;
+  cast->args = args;
+  cast->funccollid = 0;  /* funccollid */
+  cast->inputcollid = 0; /* inputcollid */
+  cast->location = location;
+  return cast;
+}
+
+static FuncExpr *make_i4tod(List *args, Oid location)
+{
+  FuncExpr *cast = make_cast_common(args, location);
+#if PG_MAJORVERSION_NUM == 13
+  cast->funcid = F_I4TOD;
+#elif PG_MAJORVERSION_NUM >= 14
+  cast->funcid = F_FLOAT8_INT4;
+#endif
+  cast->funcresulttype = FLOAT8OID;
+  return cast;
+}
+
+static FuncExpr *make_ftod(List *args, Oid location)
+{
+  FuncExpr *cast = make_cast_common(args, location);
+#if PG_MAJORVERSION_NUM == 13
+  cast->funcid = F_FTOD;
+#elif PG_MAJORVERSION_NUM >= 14
+  cast->funcid = F_FLOAT8_FLOAT4;
+#endif
+  cast->funcresulttype = FLOAT8OID;
+  return cast;
+}
+
+static FuncExpr *make_int4_numeric(List *args, Oid location)
+{
+  FuncExpr *cast = make_cast_common(args, location);
+#if PG_MAJORVERSION_NUM == 13
+  cast->funcid = F_INT4_NUMERIC;
+#elif PG_MAJORVERSION_NUM >= 14
+  cast->funcid = F_NUMERIC_INT4;
+#endif
+  cast->funcresulttype = NUMERICOID;
+  return cast;
+}
+
+static FuncExpr *make_int8_numeric(List *args, Oid location)
+{
+  FuncExpr *cast = make_cast_common(args, location);
+#if PG_MAJORVERSION_NUM == 13
+  cast->funcid = F_INT8_NUMERIC;
+#elif PG_MAJORVERSION_NUM >= 14
+  cast->funcid = F_NUMERIC_INT8;
+#endif
+  cast->funcresulttype = NUMERICOID;
+  return cast;
+}
+
+static OpExpr *make_div_operator_common(List *args, Oid location)
+{
+  OpExpr *division = makeNode(OpExpr);
+  division->opretset = false;
+  division->opcollid = 0; /* funccollid */
+  division->inputcollid = 0 /* inputcollid */,
+  division->location = location;
+  division->args = args;
+  return division;
+}
+
+static OpExpr *make_float8div(List *args, Oid location)
+{
+  OpExpr *division = make_div_operator_common(args, location);
+  division->opno = 593;
+  division->opfuncid = F_FLOAT8DIV;
+  division->opresulttype = FLOAT8OID;
+  return division;
+}
+
+static OpExpr *make_numeric_div(List *args, Oid location)
+{
+  OpExpr *division = make_div_operator_common(args, location);
+  division->opno = 1761;
```

Comment:
I didn't find a more elegant way of finding the `opno`, any ideas?

Reply:
This looks risky. I will check if there is a define for it.

Reply:
We usually lookup these Oids at startup and store them in `g_oid_cache`.

Reply:
There's `LookupOperName`, similar to `LookupFuncName`, but it seems like it requires some extra args in order to be applied here.

I couldn't find a define, but searching for `1761` has its limitations, so I might have missed something. LMK if you manage to find anything.

Reply:
I see `utils/fmgroids.h` includes `F_DIV`, `F_INT8DIV` etc. Can we emit a function call directly instead of the operator?

Reply:
Hm, looks like a different bunch of DIVs that what I looked at. Maybe we can even emit a function call, with the same function OIDs (the operator-bound ones), just without that `OpExpr` node :thinking: ?

I'll give it a try

Reply:
hm, this

>Maybe we can even emit a function call, with the same function OIDs (the operator-bound ones), just without that OpExpr node

just works, do you think there's any caveats?

Reply:
The planner is more familiar with opexpr. Also expression unparsing will not look right. But I have no idea how it pulls the OIDs for division ops..

Reply:
Maybe I'm missing what you mean by "emit a function call directly". I understand this as to remove the `OpExpr` wrapping the function, and do a `FuncExpr` instead.

I looked a bit more into it, and the `F_NUMERIC_DIV` and `F_FLOAT8DIV` I have now look like the proper functions we should use, with or without the `OpExpr` node. One can call them without the operator like `SELECT float8div(1,2)`.

Maybe I'll push a commit demonstrating the difference: d62ce0e2